### PR TITLE
GTP - Styles and Markup for articles and concepts.

### PR DESCRIPTION
**Pull request by joshuawaheed**

### CODE REVIEW

File: src/stylesheets/components/_article.scss [hunk 1]
```diff
@@ -44,4 +44,41 @@ article {
       }
     }
   }
+
+  & + aside .block .list--pipe:last-child > li:last-child {
```
Comment (UsmanAfzal) [inline src/stylesheets/components/_article.scss]: This would be useful not to be specific to article but aside as general. Can make more simple as shown below and moved into lists.scss

`aside .list--pipe li:last-child {
  border-bottom: 0;
  padding-bottom: 0;
}`

File: src/stylesheets/components/_article.scss [hunk 2]
```diff
@@ -44,4 +44,41 @@ article {
...
+    // This also fixes the extra spacing of 18px a
+    // first-child 'h2' tag makes when inside a new
+    // `section` tag. In this scenario, this fix is only
+    // happening when the `h2` tag is inside the `article`
+    // tag. However, there is now extra space above the
+    // first-child `p, ol, ul` tags to align their
+    // first-line text baselines with the `h2` tag inside
+    // the `aside` tag.
+
+    h2:first-child {
```
Comment (UsmanAfzal) [inline src/stylesheets/components/_article.scss]: Move to headings.scss

File: src/stylesheets/components/_article.scss [hunk 3]
```diff
@@ -44,4 +44,41 @@ article {
...
+      margin-top: $base-spacing-unit-medium;
+    }
+    p:first-child,
```
Comment (UsmanAfzal) [inline src/stylesheets/components/_article.scss]: As mentioned above this could be a padding on the article itself if it is consistent.

File: src/stylesheets/components/_article.scss [hunk 4]
```diff
@@ -44,4 +44,41 @@ article {
...
+    > ol:first-child > li:first-child,
```
Comment (UsmanAfzal) [inline src/stylesheets/components/_article.scss]: I know not part of this PR and it is an extension of what already has been done but is not great if the child element margins go beyond the parent. As a general pattern also no component or element except headings have a top margin. Example is `.list--block`

File: src/stylesheets/components/_article.scss [hunk 5]
```diff
@@ -44,4 +44,41 @@ article {
...
+    > ul:first-child > li:first-child {
+      margin-top: $base-spacing-unit-extra-large;
+    }
+
+    & + aside {
```
Comment (UsmanAfzal) [inline src/stylesheets/components/_article.scss]: Create and side file to manage separately and not dependant on article

File: src/stylesheets/elements/_icons.scss [hunk 6]
```diff
@@ -113,3 +117,10 @@ svg [data-icon="file"] {
     display: block;
   }
 }
+
+// Print icon
+
+[data-file-print="print"] svg {
+  vertical-align: middle;
```
Comment (UsmanAfzal) [inline src/stylesheets/elements/_icons.scss]: Can this not be applied to SVGs?

File: src/templates/prototypes/article.pug [hunk 7]
```diff
@@ -24,3 +22,38 @@ block content
               li If you are asking a substantive question and have an interest to declare, tick the box and email tableoffice@parliament.uk to say what the interest is.
           li Take your question form to the Table Office or Procedural Hub in person, or ask another MP to take it for you. Or sign the question form and post it or ask your staff to take it to the Table Office or Procedural Hub (faxed, stamped or photocopied signatures cannot be accepted).
           li The Table Office will check your question and contact you if they have a problem. Once the Table Office has approved your question, it has been tabled.
+
+      aside
+        .block.theme--grey-4
```
Comment (UsmanAfzal) [inline src/templates/prototypes/article.pug]: Would be better to extend theme--grey-4 to `aside block`

This will become the default colour, we can also override this with other themes

File: src/templates/prototypes/article.pug [hunk 8]
```diff
@@ -24,3 +22,38 @@ block content
...
+            li
+              .list--details
+                +link('#', 'MPs guide to urgent questions')
+            li
+              .list--details
+                +link('#', 'MPs guide to written questions')
+
+        .block.theme--grey-4
+          h2 Save and share
+          figure.svg--inline(data-file-print='print')
```
Comment (UsmanAfzal) [inline src/templates/prototypes/article.pug]: Better to use a definition list here. You won't need `.svg--inline` then

File: src/stylesheets/components/_article.scss (hunk 3 above)
Comment (joshuawaheed) [reply src/stylesheets/components/_article.scss]: I have removed these styles completely and have added a convenient padding instead. Also, this may look odd on pages that have a paragraph or list tag as the first child of the article tag. However, assuming it's a requirement to add a heading 2 as the first child for all articles, which we will need to address to article editors and they will have to implement, then this new update is correct.

File: src/templates/prototypes/article.pug (hunk 8 above)
Comment (joshuawaheed) [reply src/templates/prototypes/article.pug]: I've removed `vertical-align: middle;` also from the _btn_ mixin's svg element.
